wit/incoming-handler.wit

```diff
@@ -0,0 +1,21 @@
+// The `wasi:http/incoming-handler` interface is meant to be exported by
+// components and called by the host in response to a new incoming HTTP
+// response.
+//
+//   NOTE: in Preview3, this interface will be merged with
+//   `wasi:http/outgoing-handler` into a single `wasi:http/handler` interface
+//   that takes a `request` parameter and returns a `response` result.
+//
+default interface incoming-handler {
```

Comment:
Do we want to split this out into separate PRs? I feel like it's going to be a heavy lift to do both implementations in a single PR.

Of course, if it's ok to only have a partial implementation, that works too.